dags/dag_content/nextflow_tower_functions.py

```diff
@@ -28,51 +30,77 @@ def create_and_open_tower_workspace(
 
 def get_latest_compute_environment(
     tower_utils: TowerUtils,
+    compute_env_model: str,
     stack_name: str,
     workspace_id: str,
-) -> Dict[str, Optional[str]]:
-    """Gets the latest compute environments in the Nextflow Tower Workspace for a given stack
-    and workspace
+) -> str:
+    """Gets the latest compute environment in the Nextflow Tower Workspace for given stack
 
     Args:
         tower_utils (TowerUtils): Authenticated TowerUtils object that contains tower client for use
+        compute_env_model (str): Short name of the compute environment model to use
         stack_name (str): Name of the Tower stack
-        workspace_id (str): The ID number of the workspace you wish to retrieve the compute env for
+        workspace_id (str): The ID number of the workspace you wish to retrieve the env for
 
     Returns:
-        compute_env_ids (Dict): Latest SPOT and EC2 compute environments in a dictionary
+        compute_env_id (str): Latest compute environment id for the given args specified
     """
-    compute_env_ids: dict[str, Optional[str]] = {"SPOT": None, "EC2": None}
-    endpoint = "/compute-envs"
-    params = {"workspaceId": workspace_id}
-    response = tower_utils.client.request("GET", endpoint, params=params)
-
-    # figure out highest compute env version by name
+    available_envs = get_available_compute_environments(
+        tower_utils=tower_utils, workspace_id=workspace_id
+    )
+    compute_env_prefix = f"{stack_name}-{COMPUTE_ENV_MODELS[compute_env_model]}-v"
+    # gets the latest compute environment version number
     try:
-        ce_version = max(
-            [
-                int(comp_env["name"].split("-")[-1].replace("v", ""))
-                for comp_env in response["computeEnvs"]
-                if comp_env["platform"] == "aws-batch"
-                and comp_env["status"] == "AVAILABLE"
-            ]
+        max_ce_version = max(
+            list(
+                set(
+                    [
+                        int(comp_env.split("-")[-1].replace("v", ""))
+                        for comp_env in available_envs.keys()
+                        if comp_env.startswith(compute_env_prefix)
+                    ]
+                )
+            )
         )
-    except:
-        raise ValueError(
-            f"Cannot get compute environment version from Tower API request"
+    except Exception as error:
+        message = "Cannot get compute environment version from Tower API request"
+        raise ValueError(message) from error
+
+    compute_env_name = (
+        f"{stack_name}-{COMPUTE_ENV_MODELS[compute_env_model]}-v{max_ce_version}"
+    )
+
+    try:
+        compute_env_id = available_envs[compute_env_name]
+    except Exception as error:
```

Comment:
In general, you should be specific about which errors you want to handle. In this case, I'm guessing you were thinking of `KeyError`. My earlier [comment](https://github.com/Sage-Bionetworks-Workflows/orca-recipes-airflow/pull/13#discussion_r1117809287) about this was to specify **at least** `Exception`. The more specific, the better.